modules/distr-tracing-tempo-object-storage-setup-gcp-sts-install.adoc

```diff
@@ -0,0 +1,104 @@
+// Module included in the following assemblies:
+//
+//* observability/distr_tracing/distr-tracing-tempo-installing.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="setting-up-google-cloud-storage-with-security-token-service_{context}"]
+= Setting up the Google Cloud storage with the Security Token Service
+
+You can set up the Google Cloud Storage (GCS) with the Security Token Service (STS) by using the Google Cloud CLI.
+
+:FeatureName: The Amazon GCS storage with the Security Token Service
+include::snippets/technology-preview.adoc[leveloffset=+1]
+
+.Prerequisites
+
+* You have installed the latest version of the Google Cloud CLI.
+
+.Procedure
+
+. Create a GCS bucket.
+
+. Create or reuse a service account with Google's Identity and Access Management (IAM):
++
+[source,terminal]
+----
+SERVICE_ACCOUNT_EMAIL=$(gcloud iam service-accounts create <IAM_SA_NAME> \ # <1>
+    --display-name="Tempo Account" \
+    --project <PROJECT_ID>  \ # <2>
+    --format='value(email)' \ 
+    --quiet)
+----
+<1> Name of the service account in Google Cloud.
+<2> Project ID of the service account in Google Cloud.
+
+. Bind the required GCP roles to the created SA at the project level:
++
+[source,terminal]
+----
+$ gcloud projects add-iam-policy-binding <PROJECT_ID> \
+    --member "serviceAccount:$SERVICE_ACCOUNT_EMAIL" \
+  --role="roles/iam.workloadIdentityUser"
+
+$ gcloud projects add-iam-policy-binding <PROJECT_ID> \
+    --member "serviceAccount:$SERVICE_ACCOUNT_EMAIL" \
+    --role "roles/storage.objectAdmin"
+
+----
+. Retrieve the `POOL_ID` of the Google Cloud Workload Identity Pool associated with the OpenShift cluster.
++
+[source,terminal]
+----
+$ OIDC_ISSUER=$(oc get authentication.config cluster -o jsonpath='{.spec.serviceAccountIssuer}')
+
+$ POOL_ID=$(echo "$OIDC_ISSUER" | awk -F'/' '{print $NF}' | sed 's/-oidc$//')
+----
+.  Allow Kubernetes Service Accounts to impersonate the Google Service Account, SERVICE_ACCOUNT_EMAIL is the output of the commant used to created the service account in the step 2.
++
+[source,terminal]
+----
+$ gcloud iam service-accounts add-iam-policy-binding "$SERVICE_ACCOUNT_EMAIL" \
+  --role="roles/iam.workloadIdentityUser" \
+  --member="principal://iam.googleapis.com/projects/<PROJECT_NUMBER>/locations/global/workloadIdentityPools/<POOL_ID>/subject/system:serviceaccount:<TEMPO_NAMESPACE>:tempo-<TEMPO_NAME>" \
+  --project=<PROJECT_ID> \
+  --quiet
+
+$ gcloud iam service-accounts add-iam-policy-binding "$SERVICE_ACCOUNT_EMAIL" \
+  --role="roles/iam.workloadIdentityUser" \
+  --member="principal://iam.googleapis.com/projects/$PROJECT_NUMBER/locations/global/workloadIdentityPools/$POOL_ID/subject/system:serviceaccount:${TEMPO_NAMESPACE}:tempo-${TEMPO_NAME}-query-frontend" \
+  --project="$PROJECT_ID" \
+  --quiet
```

Comment:
@rubenvp8510, do we need both of these commands?
It seems to me that you added `$` in the second one.
Should I delete one of them?

Reply:
one is for query-frontend service account, so yes we need both